src/mappings/contracts.ts

```diff
@@ -246,16 +248,32 @@ export async function nodeContractCanceled(
 
   if (contractID === BigInt(0)) return
 
-  const savedContract = await ctx.store.get(NodeContract, { where: { contractID } })
+  const savedContract = await ctx.store.get(Contract, { where: { contractID } })
   if (!savedContract) return
 
   savedContract.state = ContractState.Deleted
-  await ctx.store.save<NodeContract>(savedContract)
+  await ctx.store.save<Contract>(savedContract)
+
+  let savedPublicIPs: PublicIp[] = await ctx.store.find(PublicIp, { where: { contractId: contractID }, relations: { farm: true } })
+  Promise.all(savedPublicIPs.map(async (ip) => {
+    ip.contractId = null
+    const farm = await ctx.store.get(Farm, { where: { farmID: ip.farm.farmID } })
```

Comment:
I'm not sure, but do we need to read Farm instance again? can we use `ip.farm` directly instead?